docs/core/tools/global-tools-how-to-create.md

```diff
@@ -0,0 +1,228 @@
+
+Next, add a new method named `ShowBot` that accepts a string parameter. This method will print out the message and the ascii bot. The ascii bot code was taken from the awesome [dotnetbot](https://github.com/dotnet/core/blob/master/samples/dotnetsay/Program.cs) sample.
+
+```csharp
+static void ShowBot(string message)
+{
+    string bot = $"\n        {message}";
+    bot += @"
+    __________________
+                      \
+                       \
+                          ....
+                          ....'
+                           ....
+                        ..........
+                    .............'..'..
+                 ................'..'.....
+               .......'..........'..'..'....
+              ........'..........'..'..'.....
+             .'....'..'..........'..'.......'.
+             .'..................'...   ......
+             .  ......'.........         .....
+             .    _            __        ......
+            ..    #            ##        ......
+           ....       .                 .......
+           ......  .......          ............
+            ................  ......................
+            ........................'................
+           ......................'..'......    .......
+        .........................'..'.....       .......
+     ........    ..'.............'..'....      ..........
+   ..'..'...      ...............'.......      ..........
+  ...'......     ...... ..........  ......         .......
+ ...........   .......              ........        ......
+.......        '...'.'.              '.'.'.'         ....
+.......       .....'..               ..'.....
+   ..       ..........               ..'........
+          ............               ..............
+         .............               '..............
+        ...........'..              .'.'............
+       ...............              .'.'.............
+      .............'..               ..'..'...........
+      ...............                 .'..............
+       .........                        ..............
+        .....
+";
+    Console.WriteLine(bot);
+}
+```
+
+Next, back in the `static void Main(string[] args)` method, add a call `ShowBot` as the last line of the method, passing in the message argument value.
+
+```csharp
+ShowBot(messageArg.Value);
+```
+
+### Test the tool
+
+You can run the project and see the output. Try these variations to run our application:
+
+```csharp
+dotnet run
+dotnet run -- -h
+dotnet run -- -v
+dotnet run -- "Hello from the bot"
+```
+
+All arguments after the `--` delimiter are passed to our application.
+
+## Setup the global tool
+
+Before you can pack and distribute the application as a Global Tool, you need to modify the project file. Open the `botsay.csproj` file and add three new XML nodes to the `<Project><PropertyGroup>` node:
+
+- \<PackAsTool>  
+Indicates that the application will be compiled for install as a Global Tool.
```

Comment:
to be pedantic it is packaged, not compiled. the dll is the same when you have the flag it or not